### PR TITLE
Add net8.0 target, remove obsolete, update packages

### PR DESCRIPTION
**Pull request by lahma**
Time to let go of `netoreapp3.1` and `net5.0`. `net7.0` will be obsolete soon too but left it. To sweeten the deal here are the benchmark numbers comparing jump from NET 7.0 to NET 8.0. And then for vanity comparison benchmarks under net8.0

## Runtime comparison

```

BenchmarkDotNet v0.13.10, Windows 11 (10.0.23590.1000)
AMD Ryzen 9 5950X, 1 CPU, 32 logical and 16 physical cores
.NET SDK 8.0.100
  [Host]     : .NET 7.0.14 (7.0.1423.51910), X64 RyuJIT AVX2
  Job-PXTLYP : .NET 7.0.14 (7.0.1423.51910), X64 RyuJIT AVX2
  Job-ZZEUIR : .NET 8.0.0 (8.0.23.53103), X64 RyuJIT AVX2


```
| Method         | Runtime  | Mean       | Error     | StdDev    | Ratio | Gen0   | Gen1   | Allocated | Alloc Ratio |
|--------------- |--------- |-----------:|----------:|----------:|------:|-------:|-------:|----------:|------------:|
| Parse          | .NET 7.0 |   4.915 μs | 0.0294 μs | 0.0275 μs |  1.00 | 0.1602 |      - |   2.68 KB |        1.00 |
| Parse          | .NET 8.0 |   4.636 μs | 0.0181 μs | 0.0170 μs |  0.94 | 0.1602 |      - |   2.68 KB |        1.00 |
|                |          |            |           |           |       |        |        |           |             |
| ParseBig       | .NET 7.0 |  25.978 μs | 0.0445 μs | 0.0416 μs |  1.00 | 0.7019 |      - |  11.61 KB |        1.00 |
| ParseBig       | .NET 8.0 |  24.504 μs | 0.0960 μs | 0.0851 μs |  0.94 | 0.7019 |      - |  11.61 KB |        1.00 |
|                |          |            |           |           |       |        |        |           |             |
| Render         | .NET 7.0 | 266.880 μs | 0.3826 μs | 0.2987 μs |  1.00 | 5.3711 |      - |  95.86 KB |        1.00 |
| Render         | .NET 8.0 | 200.908 μs | 1.2976 μs | 1.2138 μs |  0.75 | 5.6152 | 0.2441 |  95.86 KB |        1.00 |
|                |          |            |           |           |       |        |        |           |             |
| ParseAndRender | .NET 7.0 | 273.829 μs | 1.0554 μs | 0.9356 μs |  1.00 | 5.8594 |      - |  99.01 KB |        1.00 |
| ParseAndRender | .NET 8.0 | 206.873 μs | 0.9483 μs | 0.8406 μs |  0.76 | 5.8594 | 0.2441 |  99.01 KB |        1.00 |

## Library comparsion

```

BenchmarkDotNet v0.13.10, Windows 11 (10.0.23590.1000)
AMD Ryzen 9 5950X, 1 CPU, 32 logical and 16 physical cores
.NET SDK 8.0.100
  [Host]     : .NET 8.0.0 (8.0.23.53103), X64 RyuJIT AVX2
  DefaultJob : .NET 8.0.0 (8.0.23.53103), X64 RyuJIT AVX2


```
| Method             | Mean          | Error       | StdDev      | Ratio  | RatioSD | Gen0      | Gen1    | Gen2    | Allocated   | Alloc Ratio |
|------------------- |--------------:|------------:|------------:|-------:|--------:|----------:|--------:|--------:|------------:|------------:|
| Fluid_Parse        |      4.810 μs |   0.0900 μs |   0.0963 μs |   1.00 |    0.00 |    0.1602 |       - |       - |     2.68 KB |        1.00 |
| Scriban_Parse      |      5.084 μs |   0.1014 μs |   0.0996 μs |   1.06 |    0.03 |    0.4349 |  0.0153 |       - |     7.14 KB |        2.67 |
| DotLiquid_Parse    |      9.048 μs |   0.1742 μs |   0.2007 μs |   1.89 |    0.07 |    0.9918 |  0.0153 |       - |    16.21 KB |        6.05 |
| LiquidNet_Parse    |     36.824 μs |   0.1128 μs |   0.1000 μs |   7.66 |    0.19 |    3.7842 |  0.3052 |       - |    62.04 KB |       23.15 |
| Handlebars_Parse   |  2,916.884 μs |  14.6940 μs |  13.0259 μs | 606.94 |   13.33 |    7.8125 |  3.9063 |       - |   156.44 KB |       58.38 |
|                    |               |             |             |        |         |           |         |         |             |             |
| Fluid_ParseBig     |     25.489 μs |   0.3152 μs |   0.2632 μs |   1.00 |    0.00 |    0.7019 |       - |       - |    11.61 KB |        1.00 |
| Scriban_ParseBig   |     25.959 μs |   0.0947 μs |   0.0886 μs |   1.02 |    0.01 |    1.9531 |  0.2747 |       - |    32.06 KB |        2.76 |
| DotLiquid_ParseBig |     37.296 μs |   0.3281 μs |   0.2908 μs |   1.46 |    0.02 |    5.7373 |  0.2441 |       - |    94.36 KB |        8.13 |
| LiquidNet_ParseBig | 16,282.790 μs | 171.7901 μs | 160.6926 μs | 639.95 |   10.71 | 1718.7500 | 62.5000 |       - | 28543.39 KB |    2,458.65 |
|                    |               |             |             |        |         |           |         |         |             |             |
| Fluid_Render       |    202.840 μs |   0.4475 μs |   0.3967 μs |   1.00 |    0.00 |    5.6152 |  0.2441 |       - |    95.86 KB |        1.00 |
| Scriban_Render     |    803.347 μs |   6.0097 μs |   5.3274 μs |   3.96 |    0.03 |   68.3594 | 68.3594 | 68.3594 |   498.43 KB |        5.20 |
| DotLiquid_Render   |  1,957.585 μs |  27.8971 μs |  26.0950 μs |   9.66 |    0.13 |  203.1250 | 85.9375 | 23.4375 |  3367.85 KB |       35.13 |
| LiquidNet_Render   |  1,107.375 μs |  11.5056 μs |   9.6077 μs |   5.46 |    0.05 |  191.4063 | 82.0313 |       - |  3131.22 KB |       32.66 |
| Handlebars_Render  |    247.913 μs |   1.0110 μs |   0.8962 μs |   1.22 |    0.01 |   11.7188 |  1.9531 |       - |   194.92 KB |        2.03 |

### CODE REVIEW

**Comment by hishamco** [comment]
Agree to upgrade to .NET 8.0, but I'm not sure if @sebastienros agrees to remove the support of the previous versions

**Comment by lahma** [comment]
3.1 and 5.0 still have fallback to netstandard.

File: Fluid/Fluid.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;netstandard2.1;netcoreapp3.1;net5.0;net6.0;net7.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netstandard2.1;net6.0;net7.0;net8.0</TargetFrameworks>
```
Comment (sebastienros) [inline Fluid/Fluid.csproj:4]: I told everyone we shouldn't do that ;) Note it was already doing it before though :/

Would love to see if the benchmarks change when a specific target for net8.0 is used. Meaning run benchmarks on net8.0 but with the net8.0 specific tfm in the library, and not.

**Comment by lahma** [comment]
Here are benchmarks results when running `net8.0` benchmarks program against Fluid.csproj having `<TargetFrameworks>netstandard2.0;netstandard2.1;net6.0</TargetFrameworks>`. I don't se a reason why performance would differ as there is (currently nothing) taking advantage of conditional compilation against NET 7 or 8 APIs.

Everything seems to be within margin of error and at least as fast as before (computer has had Windows updates applied since and rebooted).

What could change generally is that for example .NET 8 SDK brought more smarts to for example `switch` against string and it could compile faster version - but this would change IL for all targets.

Generally for Fluid I'd have NET 6 and NET 8 supported as they are LTS and remove deps that netstandard versions have. `netstandard2.0` makes sense because of full framework and `netstandard2.1` as I believe for example Unity implements `netstandard2.1` (if any `netstandard2.1` condition compilation present). 

```

BenchmarkDotNet v0.13.10, Windows 11 (10.0.23595.1001)
AMD Ryzen 9 5950X, 1 CPU, 32 logical and 16 physical cores
.NET SDK 8.0.100
  [Host]     : .NET 8.0.0 (8.0.23.53103), X64 RyuJIT AVX2
  DefaultJob : .NET 8.0.0 (8.0.23.53103), X64 RyuJIT AVX2


```
| Method         | Mean       | Error     | StdDev    | Gen0   | Gen1   | Allocated |
|--------------- |-----------:|----------:|----------:|-------:|-------:|----------:|
| Parse          |   4.427 μs | 0.0171 μs | 0.0143 μs | 0.1602 |      - |   2.68 KB |
| ParseBig       |  24.262 μs | 0.0760 μs | 0.0634 μs | 0.7019 |      - |  11.61 KB |
| Render         | 205.662 μs | 1.4760 μs | 1.3807 μs | 5.6152 | 0.2441 |  95.86 KB |
| ParseAndRender | 205.381 μs | 1.0109 μs | 0.8961 μs | 5.8594 | 0.2441 |  99.01 KB |

**Comment by sebastienros** [comment]
So your argument to add net8.0 as a target for Fluid.dll is that it _shows_ it's supported. The other additions though (views, tests, benchmarks) have some technical value.

File: Fluid/Fluid.csproj (hunk 1 above)
Comment (hishamco) [reply Fluid/Fluid.csproj:4]: I told you @lahma :)

Comment (hishamco) [reply Fluid/Fluid.csproj:4]: I told you @lahma :)

**Comment by lahma** [comment]
I should probably do git blame history to find who started the pattern of supporting unnecessary targets, but feels somewhat unnecessary.

**Comment by sebastienros** [comment]
Might have been me, before I grasp when it's not required to add more. But I understand it can make users feel like it's actually note supporting net8.0 otherwise. It doesn't hurt much, just more bytes to send. It's also adding the tfm just in case we need to add custom code someday.

**Review by sebastienros — APPROVED** [review]

**Pull request merged**